winml/lib/Api.Core/inc/WinMLAdapter.h

```diff
@@ -131,11 +121,14 @@ class InferenceSession : public Microsoft::WRL::RuntimeClass <
     InferenceSession(onnxruntime::InferenceSession * session);
 
     onnxruntime::InferenceSession* STDMETHODCALLTYPE get() override { return session_.get(); }
+    HRESULT STDMETHODCALLTYPE GetOrtSession(OrtSession ** out) override {
+        // (OrtSession *) are really (InferenceSession *) as well
+        *out = reinterpret_cast<OrtSession*>(session_.get());
```

Comment:
>reinterpret_cast [](start = 15, length = 16)

do we need to do a reinterpret_cast? I dream of a world where we no longer use RTTI, so I'm hesitant to add new dependencies on it

Reply:
I think only `dynamic_cast` uses RTTI

Reply:
`reinterpret_cast` just reinterprets the pointer type (similar to a raw C-style cast).

Reply:
yup, my bad. I get those confused sometimes

---
In reply to: [351498594](https://github.com/microsoft/onnxruntime/pull/2504#discussion_r351498594) [](ancestors = 351498594)